bolt-design-system/bolt
# Build Tools | Dev | Update Browsersync Config To Allow Customizable Proxy Url

## Jira

http://vjira2:8080/browse/BDS-1240

## Summary

Give the possibility to specify the proxy hostname and proxy port via config.

## Details

Update configuration and how was previous proxy handled. 

## How to test

1. Run this code locally.
![image](https://user-images.githubusercontent.com/16049049/60659372-81498a80-9e55-11e9-9f9d-c4d77508bc7d.png)

1. First, check if our `docs-site` is working correctly website is rendering correctly and all auto-reload is working as expected (HTML, CSS, JS and TWIG).
1. After this check please check your IP address (in my example is `192.168.1.10`)
1. Run simple PHP server by running this command in terminal `php -S YOUR_IP_ADDRESS:PROXY_PORT -t www/` - change YOUR_IP_ADDRESS with your IP address and PROXY_PORT by adding specific port. _this command must be run from root of the project_
![image](https://user-images.githubusercontent.com/16049049/60658946-835f1980-9e54-11e9-8a0e-3361aafb8821.png)

1. Check on browser if on passing YOUR_IP_ADDRESS and PROXY_PORT you see our bolt website.
![image](https://user-images.githubusercontent.com/16049049/60659023-ac7faa00-9e54-11e9-9b81-49e9a627b5ab.png)

1. Update `.boltrc.js` localized in `docs-site` folder by adding two lines of code 
`proxyHostname: 'YOUR_IP_ADDRESS',
  proxyPort: PROXY_PORT,`
(in my example it looks like this: `proxyHostname: '192.168.1.10', proxyPort: 8810,`
![image](https://user-images.githubusercontent.com/16049049/60659055-c91be200-9e54-11e9-8ec2-e389d6e981bb.png)

1. Run `npm start` as usual from the root folder.
![image](https://user-images.githubusercontent.com/16049049/60659428-ad650b80-9e55-11e9-9f27-14616e143dff.png)
1. Navigate to `localhost:300` and compare if changes made in our code is reflecting on this website and after reload in port and hostname provided in config file.
